packages/eas-cli/src/credentials/ios/appstore/provisioningProfileAdhoc.ts

```diff
@@ -1,9 +1,219 @@
+import {
+  Certificate,
+  CertificateType,
+  Device,
+  Profile,
+  ProfileState,
+  ProfileType,
+  Teams,
+} from '@expo/apple-utils';
 import ora from 'ora';
 
 import { ProvisioningProfile } from './Credentials.types';
 import { AuthCtx } from './authenticate';
+import { USE_APPLE_UTILS, getBundleIdForIdentifier, getProfilesForBundleId } from './experimental';
 import { runActionAsync, travelingFastlane } from './fastlane';
 
+async function registerMissingDevicesAsync(udids: string[]) {
+  const allIosProfileDevices = await Device.getAllIOSProfileDevicesAsync();
+  const alreadyAdded = allIosProfileDevices.filter(d => udids.includes(d.attributes.udid));
+  const alreadyAddedUdids = alreadyAdded.map(i => i.attributes.udid);
+
+  for (const udid of udids) {
+    if (!alreadyAddedUdids.includes(udid)) {
+      const device = await Device.createAsync({ name: 'iOS Device (added by Expo)', udid });
+      alreadyAdded.push(device);
+    }
+  }
+
+  return alreadyAdded;
+}
+
+async function findDistCert(serial_number: string, isEnterprise: boolean) {
```

Comment:
```suggestion
async function findDistCertAsync(serialNumber: string, isEnterprise: boolean) {
```
We don't use snake_case in js.